### ISSUE TITLE
Relative imports in a submodule are unresolved when there are symlinks

### ISSUE THREAD

**Issue by ipkiss42** [microsoft/pyright#10953]
**Describe the bug**

The issue can be reproduced using the following script:

```sh
#!/bin/sh
mkdir repro && cd repro || exit 1
base=$(pwd)
mkdir -p src/foo/bar other/foo/bar

echo '{ "executionEnvironments": [{"root": "src"}], "pythonVersion": "3.9"}' > pyrightconfig.json

cat > src/main.py << EOF
from foo import Baz
from foo.bar import Baz as Baz2

baz1 = Baz()
baz2 = Baz2()
EOF

cat > other/foo/__init__.py << EOF
from foo.bar import Baz

__all__ = ["Baz"]
EOF

cat > other/foo/bar/__init__.py << EOF
# from foo.bar.baz import Baz
from .baz import Baz

__all__ = ["Baz"]
EOF

cat > other/foo/bar/baz.py << EOF
class Baz:
    pass
EOF

ln -s "$base/other/foo/__init__.py" "src/foo/__init__.py"
ln -s "$base/other/foo/bar/__init__.py" "src/foo/bar/__init__.py"
ln -s "$base/other/foo/bar/baz.py" "src/foo/bar/baz.py"
```

The script sets up a few python scripts, but apart from `src/main.py` they are all symlinks to other files (following the same directory structure).

In the `repro` directory, running `python src/main.py` works but `pyright .` fails:

```sh
> pyright .
/tmp/repro/other/foo/__init__.py
  /tmp/repro/other/foo/__init__.py:1:21 - error: "Baz" is unknown import symbol (reportAttributeAccessIssue)
/tmp/repro/src/foo/__init__.py
  /tmp/repro/src/foo/__init__.py:1:21 - error: "Baz" is unknown import symbol (reportAttributeAccessIssue)
/tmp/repro/src/foo/bar/__init__.py
  /tmp/repro/src/foo/bar/__init__.py:2:6 - error: Import ".baz" could not be resolved (reportMissingImports)
/tmp/repro/src/main.py
  /tmp/repro/src/main.py:1:17 - error: "Baz" is unknown import symbol (reportAttributeAccessIssue)
  /tmp/repro/src/main.py:2:21 - error: "Baz" is unknown import symbol (reportAttributeAccessIssue)
5 errors, 0 warnings, 0 informations
```

However, if you replace the relative import with an absolute one in `src/foo/bar/__init__.py` (see commented line in that file), Pyright does not find errors anymore.

Note that using regular files instead of symlinks also works, so they definitely play a role here.

You may be wondering why I have this strange setup with symlinks. They are due to a custom build tool at my workplace, which aggregates various source packages. There is some caching involved, so files from all packages are downloaded to a central place, and venvs are mostly made of symlinks...

**Code or Screenshots**
See above

**VS Code extension or command-line**
Using Pyright 1.1.405 from the command-line

**Comment by erictraut** [comment]
Do you happen to know whether this behavior is a regression in 1.1.405? There was a regression related to symlinks that has [already been fixed](https://github.com/microsoft/pyright/issues/10906).

**Comment by ipkiss42** [comment]
Yes, Pyright 1.1.404 works fine! Sorry for the duplicate issue; I'm resolving it now.

**Comment by erictraut** [comment]
Thanks for confirming. I'm glad to hear this is fixed.
